### PR TITLE
Implement AVX/AVX2/SSE3 Load* intrinsics

### PR DESCRIPTION
**Pull request by fiigii**
@CarolEidt @tannergooding PTAL

### CODE REVIEW

**Comment by fiigii** [comment]
test Windows_NT x64 Checked jitincompletehwintrinsic
test Windows_NT x64 Checked jitx86hwintrinsicnoavx
test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
test Windows_NT x64 Checked jitx86hwintrinsicnosimd
test Windows_NT x64 Checked jitnox86hwintrinsic

test Windows_NT x86 Checked jitincompletehwintrinsic
test Windows_NT x86 Checked jitx86hwintrinsicnoavx
test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
test Windows_NT x86 Checked jitx86hwintrinsicnosimd
test Windows_NT x86 Checked jitnox86hwintrinsic

test Ubuntu x64 Checked jitincompletehwintrinsic
test Ubuntu x64 Checked jitx86hwintrinsicnoavx
test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
test Ubuntu x64 Checked jitx86hwintrinsicnosimd
test Ubuntu x64 Checked jitnox86hwintrinsic

test OSX10.12 x64 Checked jitincompletehwintrinsic
test OSX10.12 x64 Checked jitx86hwintrinsicnoavx
test OSX10.12 x64 Checked jitx86hwintrinsicnoavx2
test OSX10.12 x64 Checked jitx86hwintrinsicnosimd
test OSX10.12 x64 Checked jitnox86hwintrinsic

File: src/jit/hwintrinsiclistxarch.h [hunk 1]
```diff
@@ -173,6 +175,7 @@ HARDWARE_INTRINSIC(SSSE3_IsSupported,                                "get_IsSupp
...
 HARDWARE_INTRINSIC(SSE41_BlendVariable,                              "BlendVariable",                                    SSE41,      -1,           16,           3,           {INS_pblendvb,  INS_pblendvb,  INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_invalid,   INS_blendvps,  INS_blendvpd},          HW_Category_SimpleSIMD,                        HW_Flag_NoFlag)
+HARDWARE_INTRINSIC(SSE41_LoadAlignedVector128NonTemporal,            "LoadAlignedVector128NonTemporal",                           SSE41,      -1,           16,           1,           {INS_movntdqa,  INS_movntdqa,  INS_movntdqa,  INS_movntdqa,  INS_movntdqa,  INS_movntdqa,  INS_movntdqa,  INS_movntdqa,  INS_invalid,   INS_invalid},           HW_Category_MemoryLoad,                        HW_Flag_NoFlag)
```
Comment (tannergooding) [inline src/jit/hwintrinsiclistxarch.h]: nit: alignment of fields is off

Comment (fiigii) [reply src/jit/hwintrinsiclistxarch.h]: Thanks, will fix.

**Comment by tannergooding** [comment]
It would be useful to uncomment the `ProcessInputs` lines for `Avx` and `Avx2` and to rerun the script: https://github.com/dotnet/coreclr/blob/master/tests/src/JIT/HardwareIntrinsics/X86/Shared/GenerateTests.csx#L169

The only thing blocking them (to my knowledge) was the `LoadVector256` and `LoadAlignedVector256` implementations.

**Comment by tannergooding** [comment]
Are you going to handle the `Store` intrinsics in a separate PR?

**Comment by fiigii** [comment]
> Are you going to handle the Store intrinsics in a separate PR?

Yes, I will submit the store PR tomorrow.

**Comment by 4creators** [comment]
> Yes, I will submit the store PR tomorrow.
@fiigii will it comprise Sse2 as well?

**Comment by fiigii** [comment]
> will it comprise Sse2 as well?

Yes, I will implement all Sse2.Store*.

**Comment by fiigii** [comment]
@CarolEidt @tannergooding RyuJIT emitter seems not support SSE4.1 `movntdqa` without VEX-encoding. I removed `Sse41.LoadAlignedVector128NonTemporal` from this PR and logged the issue at https://github.com/dotnet/coreclr/issues/16216. 

Can we merge these `Load` intrinsics at first and implement `Sse41.LoadAlignedVector128NonTemporal` later? That would be good to test containment on `Load*`.

**Comment by fiigii** [comment]
test Windows_NT x64 Checked jitincompletehwintrinsic
test Windows_NT x64 Checked jitx86hwintrinsicnoavx
test Windows_NT x64 Checked jitx86hwintrinsicnoavx2
test Windows_NT x64 Checked jitx86hwintrinsicnosimd
test Windows_NT x64 Checked jitnox86hwintrinsic

test Windows_NT x86 Checked jitincompletehwintrinsic
test Windows_NT x86 Checked jitx86hwintrinsicnoavx
test Windows_NT x86 Checked jitx86hwintrinsicnoavx2
test Windows_NT x86 Checked jitx86hwintrinsicnosimd
test Windows_NT x86 Checked jitnox86hwintrinsic

test Ubuntu x64 Checked jitincompletehwintrinsic
test Ubuntu x64 Checked jitx86hwintrinsicnoavx
test Ubuntu x64 Checked jitx86hwintrinsicnoavx2
test Ubuntu x64 Checked jitx86hwintrinsicnosimd
test Ubuntu x64 Checked jitnox86hwintrinsic

test OSX10.12 x64 Checked jitincompletehwintrinsic
test OSX10.12 x64 Checked jitx86hwintrinsicnoavx
test OSX10.12 x64 Checked jitx86hwintrinsicnoavx2
test OSX10.12 x64 Checked jitx86hwintrinsicnosimd
test OSX10.12 x64 Checked jitnox86hwintrinsic

**Comment by fiigii** [comment]
@tannergooding Do you know why do these OSX tests fail? There is no error message.

**Comment by tannergooding** [comment]
Looks like the machines had network issues. I have requeued the jobs.

**Comment by fiigii** [comment]
Thank you. Looks like unrelated failures. 
Rebased to solve the conflict from SSE2 load.

**Comment by fiigii** [comment]
@CarolEidt Does this PR look good to you?

**Comment by CarolEidt** [comment]
> It would be useful to uncomment the ProcessInputs lines for Avx and Avx2 and to rerun the script

It would be good to know if you have done that. If not, could you open an issue to enable the template for those?

**Comment by fiigii** [comment]
@tannergooding @CarolEidt Thank for the review, I logged the containment test work at https://github.com/dotnet/coreclr/issues/16244. Will submit a PR soon.

**Comment by fiigii** [comment]
@CarolEidt May I have the permission to add PR/issues to "Hardware Intrinsic Project" if possible? That may save your time 😄

**Pull request merged**